### PR TITLE
Fixes #6357: File directory

### LINKED ISSUE #6357: General file directory ignored
Dev version JabRef 5.1--2020-04-23--bbea7df

The General file directory (specified in Library Properties) is ignored. Files are now saved by default in the same folder as the bib file, and already linked files that are placed in the General file directory are no longer found.

@calixtus could this be a result of your recent refatorings?

(Sorry don't have time right now to investigate).

### PR DESCRIPTION
**Pull request by tobiasdiez**
Bug was introduced in https://github.com/JabRef/jabref/commit/1b03f039aa4abf0812a3a8ae0ba248c08fdb39a3. Sorry @calixtus for suspecting you 👼.

Also a bit of code cleanup is included. For example, I tried to minimize using strings and favored `Path`s instead.

- [ ] Change in CHANGELOG.md described (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/model/metadata/FilePreferences.java [hunk 1]
```diff
@@ -34,24 +30,14 @@ public String getUser() {
...
-                Path path = Paths.get(value);
-                return Optional.of(path);
-            }
-            return Optional.empty();
-        } catch (InvalidPathException ex) {
+    public Optional<Path> getFileDirectory() {
+        if (StringUtil.isBlank(mainFileDirectory)) {
             return Optional.empty();
+        } else {
+            return Optional.of(Paths.get(mainFileDirectory));
```
Comment (calixtus) [inline src/main/java/org/jabref/model/metadata/FilePreferences.java:37]: JavaDoc indicates, that `Paths::get` will be deprecated soon. Use `Path::of` instead.
https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/nio/file/Paths.html

File: src/test/java/org/jabref/model/database/BibDatabaseContextTest.java [hunk 2]
```diff
@@ -15,61 +17,104 @@
...
     @Test
-    public void getFileDirectoriesWithEmptyDbParent() {
-        BibDatabaseContext dbContext = new BibDatabaseContext();
-        dbContext.setDatabasePath(Paths.get("biblio.bib"));
-        List<String> fileDirectories = dbContext.getFileDirectories(StandardField.FILE, fileDirPrefs);
-        assertEquals(Collections.singletonList(currentWorkingDir.toString()),
-                fileDirectories);
+    void getFileDirectoriesWithEmptyDbParent() {
+        BibDatabaseContext database = new BibDatabaseContext();
+        database.setDatabasePath(Paths.get("biblio.bib"));
```
Comment (calixtus) [inline src/test/java/org/jabref/model/database/BibDatabaseContextTest.java:36]: `Path::of` and also below

**Comment by calixtus** [comment]
> Sorry @calixtus for suspecting you 👼.
Forgiven and forgotten. You owe me one. 😉

**Comment by tobiasdiez** [comment]
Thanks for the quick reviews!

**Pull request merged**